src/libsystemd/sd-network/network-util.c

```diff
@@ -56,3 +56,54 @@ static const char* const link_address_state_table[_LINK_ADDRESS_STATE_MAX] = {
 };
 
 DEFINE_STRING_TABLE_LOOKUP(link_address_state, LinkAddressState);
+
+static int parse_operational_state(const char *str) {
+        LinkOperationalState s;
+
+        if (isempty(str))
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                        "Operational state is empty.");
+
+        s = link_operstate_from_string(str);
+        if (s < 0)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                        "Invalid operational state '%s'", str);
+
+        return s;
+}
+
+int parse_operational_state_pair(const char *str, LinkOperationalStatePair *out) {
+        LinkOperationalStatePair state;
+        _cleanup_free_ const char *min = NULL;
+        const char *p;
+
+        assert(str);
+        assert(out);
+
+        p = strchr(str, ':');
+        if (p) {
+                min = strndup(str, p - str);
+
+                state.max = parse_operational_state(p + 1);
+                if (state.max < 0)
+                        return state.max;
+        } else {
+                min = strdup(str);
+                state.max = _LINK_OPERSTATE_INVALID;
+        }
+
+        if (!min)
+                return log_oom();
+
+        if (isempty(min + 1))
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                        "Minimum operational state is empty.");
```

Comment:
Hmm? Sorry, I cannot understand this block.

Reply:
Seems I forgot to remove this when I extracted `parse_operational_state` to a separate function.